src/test/java/edu/rice/pcdp/finish/TestFinish9TooMoreFinish.java

```diff
@@ -0,0 +1,80 @@
+package edu.rice.pcdp.finish;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static edu.rice.pcdp.PCDP.async;
+import static edu.rice.pcdp.PCDP.finish;
+
+/**
+ * There are no  dead lock and blocked workers
+ * when execution task haves quantity of finish
+ * statements grater then quantity of workers.
+ *
+ * This test does not proof correct invocation
+ * in concurrency environment.
+ *
+ * @author BadPitt
+ */
+@RunWith(JUnit4.class)
+public class TestFinish9TooMoreFinish {
+
+    private static final int ARRAY_LENGTH = 10_000;
+    private static final int[] ARRAY = new int[ARRAY_LENGTH];
+
+    @BeforeClass
+    public static void setUp() {
+        for (int i = 0; i < ARRAY.length; i++) {
+            ARRAY[i] = i+1;
+        }
+    }
+
+    @Test
+    public void nestedFinishTest() {
+        final int dataLength = Runtime.getRuntime().availableProcessors() * 2;
+        final int[] parResult =  new int[1];
+
+        finish(() -> {
+            parResult[0] = sum(dataLength, 0, 0, ARRAY.length);
+        });
+        System.out.printf("Parallel result is: %d\n", parResult[0]);
+
+        int seqResult =  0;
+        for (int i = 0; i < ARRAY.length; i++) {
+            seqResult += ARRAY[i];
+        }
+        System.out.printf("Sequencial result is: %d", seqResult);
```

Comment:
We should include an assertion on the equality of the sequential and parallel sums.